File: 274. H-Index.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function(citations) {
    n = citations.length;
    citations.sort(function(a,b){return a-b});
    for (var i = 0; i < n; i++)
    {
        if (citations[i] >= n-i)
            return n-i;
    }
    return 0;
};

File: 274. H-Index.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        n = len(citations)
        citations = sorted(citations)
        for i in xrange(n):
            if n-i <= citations[i]:
                return n-i
        return 0
